### ISSUE TITLE
[Feature Request] Allow a custom Options Layout

### ISSUE THREAD

**Issue by SitiSchu** [c-blake/cligen#11]
Right now the layout of the option description is something like
`$long, $short     (toggle|string)  (""|false)  $desc`

I'd prefer if it was possible to change it to something like
`$short, $long   $desc`

Its already possible to customize the usage param so it would be nice if it was possible to customize the options layout too.

**Comment by c-blake** [comment]
I just did a push to (almost) address this which was a TODO item of my own.  See test/HelpTabCols.nim for usage.  The "almost" comes from not being able to re-order short & long in the keys column as you were asking.  From the help table perspective, that is just a single keys column.  It was not clear from your request how much reversing short and long vs dropping type and default mattered to you.  So, this recent feature add alone may satisfy you.

If it does not, that column is set up in the `argcvt.keys` proc and might be made user-overridable or tunable, though we may need to take some lambda defaulting to argcvt.keys to wire that into the system or take yet another helpTab argument like a `helpTabKeyOrder` or something to pass to `keys()`.  Personally, I think it should always be long followed by short because a long option key matching the Nim proc parameter is mandatory while a short option key is not mandatory.  That said, formatting tastes can still differ quite reasonably and a lot depends on what one is used to seeing.

I've held off on doing some full table formatting language for the help table that also auto-aligns/wraps various columns.  That seems more like a new, separate library in its best/most general form, and I wanted to keep cligen's dependencies minimal.

I am closing this issue.  If I haven't persuaded you as to long,short being the one true order, open a new one about making keys() better/overridable.  I consider that distinct from the table layout which is pretty flexible now.

**Comment by c-blake** [comment]
Also, I did a highly unscientific study of all 36 programs in my `/usr/bin` that dump a help message with `--help` and have verbose/v options.  Your short, long order is an 80% favorite vs 20% long,short.  Given that popularity study, I switched the order in `argcvt.keys()` to what you were asking for.  It would not be hard to do a `keysFmt` string someday if anyone else seems to not like short,long or maybe wants to relocate the '=' or use only one of them (like `-a, --alpha=`) or use `':'` or space instead of `=`, etc.
